### ISSUE TITLE
[Research] Add DreamNet NLP Dream Journal Analysis (92% theme classification)

### ISSUE THREAD

**Issue by LakshmiSravyaVedantham** [LakshmiSravyaVedantham/DreamAnalysis#48]
## What

Add NLP-based dream theme classification to the dream journal feature. DreamNet (arXiv 2503.05778, 2025) achieves 92.1% accuracy on dream theme classification from text reports, and 99.0% with EEG fusion.

## Why

- Current dream analysis uses GPT for narrative analysis — good but unstructured
- DreamNet classifies dream themes: flying, falling, pursuit, loss, etc.
- Text-only: 92.1% accuracy, 88.4% F1
- With REM-stage EEG fusion: 99.0% accuracy (+7% gain)
- Correlations discovered: falling-anxiety r=0.91
- Creates closed loop: EEG detects REM → user journals dream → system classifies themes → correlates with EEG patterns

### Step 1: Dream theme classifier
```python
# ml/models/dream_nlp_classifier.py (NEW)
class DreamThemeClassifier:
    """Classify dream themes from text reports.
    Uses fine-tuned DistilBERT for theme + emotion classification."""
    
    THEMES = ["flying", "falling", "pursuit", "loss", "social", "exploration"]
    EMOTIONS = ["fear", "joy", "anxiety", "sadness", "wonder"]
    
    def classify(self, dream_text: str) -> dict:
        """Classify dream themes and emotional content."""
```

### Step 2: API endpoint
- `POST /dream-analysis/classify` — returns theme probabilities + emotional analysis
- Complements existing `/api/dream-analysis` (GPT narrative)

### Step 3: EEG-dream correlation
- Store classified themes in database
- Cross-reference with EEG sleep stage data from same night
- Build "Dream Patterns" dashboard showing theme frequency over time

### Files to Modify
- `ml/models/dream_nlp_classifier.py` (NEW) — Theme classification
- `ml/api/routes.py` — Add `/dream-analysis/classify` endpoint
- `client/src/pages/dream-journal.tsx` — Show classified themes

### Expected Impact
- 92%+ dream theme classification
- Structured, quantified dream analysis
- EEG-dream correlation insights

## Complexity
MEDIUM — DistilBERT fine-tuning, needs dream report training data

## References
- [DreamNet](https://arxiv.org/abs/2503.05778) — 2025, 1,500 narratives, 92.1% text-only
- DREAM Database (Nature Communications, 2025) — 505 subjects, 2,643 awakenings
